modules/programs/kubeswitch.nix

```diff
@@ -0,0 +1,120 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+let
+  cfg = config.programs.kubeswitch;
+
+  yamlFormat = pkgs.formats.yaml { };
+in
+{
+  meta.maintainers = [ lib.hm.maintainers.m0nsterrr ];
+
+  options.programs.kubeswitch = {
+    enable = lib.mkEnableOption "the kubectx for operators";
+
+    commandName = lib.mkOption {
+      type = lib.types.str;
+      default = "kswitch";
+      description = "The name of the command to use";
+    };
+
+    package = lib.mkPackageOption pkgs "kubeswitch" { };
+
+    enableBashIntegration = lib.hm.shell.mkBashIntegrationOption { inherit config; };
+
+    enableFishIntegration = lib.hm.shell.mkFishIntegrationOption { inherit config; };
+
+    enableZshIntegration = lib.hm.shell.mkZshIntegrationOption { inherit config; };
+
+    settings = lib.mkOption {
+      type = yamlFormat.type;
+      default = { };
+      defaultText = lib.literalExpression "{ }";
+      example = lib.literalExpression ''
+        {
+          kind = "SwitchConfig";
+          kubeconfigName = "*.myconfig";
+          kubeconfigStores = [
+            {
+              kind = "filesystem";
+              kubeconfigName = "*.myconfig";
+              paths = [
+                "~/.kube/my-other-kubeconfigs/"
+              ];
+            }
+          ];
+          version = "v1alpha1";
+        }
+      '';
+      description = ''
+        Configuration written to
+        {file}`~/.kube/switch-config.yaml`.
+      '';
+    };
+  };
+
+  config = lib.mkIf cfg.enable (
+    lib.mkMerge [
+      {
+        home.packages = [ cfg.package ];
+
+        home.file = {
+          ".kube/switch-config.yaml" = lib.mkIf (cfg.settings != { }) {
+            source = yamlFormat.generate "kubeswitch-settings" cfg.settings;
+          };
+        };
+      }
+
+      (lib.mkIf cfg.enableBashIntegration (
+        let
+          shell_files = pkgs.runCommand "kubeswitch-shell-files" { buildInputs = [ cfg.package ]; } ''
+            mkdir -p $out/share
+            switcher init bash | sed 's/switch(/${cfg.commandName}(/' > $out/share/${cfg.commandName}_init.bash
+            switcher --cmd ${cfg.commandName} completion bash > $out/share/${cfg.commandName}_completion.bash
+          '';
```

Comment:
Would be nice to make all these repeated blocks a single function that generates from a `shell` input. 

```nix
generateKubeswitchShellFiles = shell:
    pkgs.runCommand "kubeswitch-${cfg.commandName}-shell-files-for-${shell}" {
      nativeBuildInputs = [ cfg.package ];
      inherit shell;
      commandName = cfg.commandName;
    } ''
      mkdir -p $out/share
      switcher init "$shell" | sed "s/switch(/$commandName(/" > "$out/share/${commandName}_init.$shell"
      switcher --cmd "$commandName" completion "$shell" > "$out/share/${commandName}_completion.$shell"
    '';
```

Then you could just have a single variable in each shell 

```nix
  kubeswitchBashFiles = generateKubeswitchShellFiles "bash";
```

Reply:
No problem :)
Thanks for your help with my first PR related to Nix*.
I didn’t factor out the Fish snippet as it’s a little bit different from Bash and Zsh

Reply:
What's different about fish? Looking at it it appears to be the same code